// @cyntler/react-doc-viewer/dist/cjs/store/actions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setMainConfig = exports.SET_MAIN_CONFIG = exports.setRendererRect = exports.SET_RENDERER_RECT = exports.updateCurrentDocument = exports.UPDATE_CURRENT_DOCUMENT = exports.previousDocument = exports.PREVIOUS_DOCUMENT = exports.nextDocument = exports.NEXT_DOCUMENT = exports.setDocumentLoading = exports.SET_DOCUMENT_LOADING = exports.setAllDocuments = exports.SET_ALL_DOCUMENTS = void 0;
exports.SET_ALL_DOCUMENTS = "SET_ALL_DOCUMENTS";
var setAllDocuments = function (documents, initialActiveDocument) { return ({
    type: exports.SET_ALL_DOCUMENTS,
    documents: documents,
    initialActiveDocument: initialActiveDocument,
}); };
exports.setAllDocuments = setAllDocuments;
exports.SET_DOCUMENT_LOADING = "SET_DOCUMENT_LOADING";
var setDocumentLoading = function (value) { return ({
    type: exports.SET_DOCUMENT_LOADING,
    value: value,
}); };
exports.setDocumentLoading = setDocumentLoading;
exports.NEXT_DOCUMENT = "NEXT_DOCUMENT";
var nextDocument = function () { return ({ type: exports.NEXT_DOCUMENT }); };
exports.nextDocument = nextDocument;
exports.PREVIOUS_DOCUMENT = "PREVIOUS_DOCUMENT";
var previousDocument = function () { return ({
    type: exports.PREVIOUS_DOCUMENT,
}); };
exports.previousDocument = previousDocument;
exports.UPDATE_CURRENT_DOCUMENT = "UPDATE_CURRENT_DOCUMENT";
var updateCurrentDocument = function (document) { return ({ type: exports.UPDATE_CURRENT_DOCUMENT, document: document }); };
exports.updateCurrentDocument = updateCurrentDocument;
exports.SET_RENDERER_RECT = "SET_RENDERER_RECT";
var setRendererRect = function (rect) { return ({
    type: exports.SET_RENDERER_RECT,
    rect: rect,
}); };
exports.setRendererRect = setRendererRect;
exports.SET_MAIN_CONFIG = "SET_MAIN_CONFIG";
var setMainConfig = function (config) { return ({
    type: exports.SET_MAIN_CONFIG,
    config: config,
}); };
exports.setMainConfig = setMainConfig;
